source/stdx/allocator/building_blocks/free_list.d

```diff
@@ -782,21 +777,23 @@ struct SharedFreeList(ParentAllocator,
         }
         @property void min(size_t x) shared
         {
-            enforce(x <= max);
-            enforce(cas(&_min, chooseAtRuntime, x),
-                "SharedFreeList.min must be initialized exactly once.");
+            if (!(x <= max))
+                throw excBounds;
+            if (!(cas(&_min, chooseAtRuntime, x)))
+                throw excMin;
         }
         static if (maxSize == chooseAtRuntime)
         {
             // Both bounds can be set, provide one function for setting both in
             // one shot.
             void setBounds(size_t low, size_t high) shared
             {
-                enforce(low <= high && high >= (void*).sizeof);
-                enforce(cas(&_min, chooseAtRuntime, low),
-                    "SharedFreeList.min must be initialized exactly once.");
-                enforce(cas(&_max, chooseAtRuntime, high),
-                    "SharedFreeList.max must be initialized exactly once.");
+                if (!(low <= high && high >= (void*).sizeof))
+                    throw excBounds;
```

Comment:
Why not add an enforce template that will create the static exception?

Reply:
Maybe will add it in the future. Mir Exception is declared in mir-runtime, need to play with DIP1008 more.

Reply:
I have used such enforce in the production. How would you name it?

Reply:
I would actually use `enforce`, but probably e.g. `massert` would work too.
What's more interesting is how such a template would be. A few ideas of my head:

a) `enforce!"SharedFreeList.max must .."(!(low <= high && high >= (void*).sizeof))` (pass exception string as template paramter)
b) `enforce!"SharedFreeList"(!(low <= high && high >= (void*).sizeof), "SharedFreeList.max must ..")` (pass unique id as template parameter + custom exception string)
c) `enforce(!(low <= high && high >= (void*).sizeof), "SharedFreeList.max must ..")` (uses function caller name as automatic template argument)